src/main/java/uk/gov/justice/hmpps/casenotes/services/CaseNoteService.java

```diff
@@ -40,36 +44,85 @@ public CaseNoteService(OffenderCaseNoteRepository repository, CaseNoteTypeReposi
-        final var filter = OffenderCaseNoteFilter.builder()
-                .offenderIdentifier(offenderIdentifier)
-                .build();
+    public Page<CaseNote> getCaseNotes(final String offenderIdentifier, final CaseNoteFilter caseNoteFilter, final Pageable pageable) {
 
-        return getCaseNotes(pageable, filter);
-    }
+        final var sensitiveCaseNotes = new ArrayList<CaseNote>();
 
-    public Page<CaseNote> getCaseNotes(final CaseNoteFilter caseNoteFilter, final Pageable pageable) {
-        final var filter = OffenderCaseNoteFilter.builder()
-                .type(caseNoteFilter.getType())
-                .subType(caseNoteFilter.getSubType())
-                .locationId(caseNoteFilter.getLocationId())
-                .staffUsername(caseNoteFilter.getStaffUsername())
-                .startDate(caseNoteFilter.getStartDate())
-                .endDate(caseNoteFilter.getEndDate())
-                .build();
+        if (SecurityUserContext.hasRoles("VIEW_SENSITIVE_CASE_NOTES", "ADD_SENSITIVE_CASE_NOTES")) {
 
-        return getCaseNotes(pageable, filter);
-    }
+            final var filter = OffenderCaseNoteFilter.builder()
+                    .offenderIdentifier(offenderIdentifier)
+                    .type(caseNoteFilter.getType())
+                    .subType(caseNoteFilter.getSubType())
+                    .locationId(caseNoteFilter.getLocationId())
+                    .staffUsername(caseNoteFilter.getStaffUsername())
+                    .startDate(caseNoteFilter.getStartDate())
+                    .endDate(caseNoteFilter.getEndDate())
+                    .build();
+
+            sensitiveCaseNotes.addAll(repository.findAll(filter)
+                    .stream()
+                    .map(this::mapper)
+                    .collect(Collectors.toList()));
+        }
+
+        // only supports one field sort.
+        final var direction = pageable.getSort().isSorted() ? pageable.getSort().get().map(Sort.Order::getDirection).collect(Collectors.toList()).get(0) : Sort.Direction.DESC;
+        final var sortField = pageable.getSort().isSorted() ? pageable.getSort().get().map(Sort.Order::getProperty).collect(Collectors.toList()).get(0) : "occurrenceDateTime";
+
+        Page<CaseNote> caseNotes;
+        if (sensitiveCaseNotes.isEmpty()) {
+            // Just degate to elite2 for data
+            final var pagedNotes = externalApiService.getOffenderCaseNotes(offenderIdentifier, caseNoteFilter, pageable.getPageSize(), pageable.getPageNumber(), sortField, direction);
+
+            final var dtoNotes = translateToDto(pagedNotes, offenderIdentifier);
+            caseNotes = new PageImpl<>(dtoNotes, pageable, pagedNotes.getTotalElements());
+
+        } else {
+            // There are both case note sources.  Combine
+            final var pagedNotes = externalApiService.getOffenderCaseNotes(offenderIdentifier, caseNoteFilter, 1000, 0, "caseNoteId", ASC);
```

Comment:
this seems quite naff to get all the data.  Could we get away with retrieving `pageSize + offset + secureCaseNotesCount` records instead?

Reply:
Of course it would be much simpler to treat them both as separate types of case notes and then we wouldn't need to combine...

Reply:
Hmm can we be sure we are getting the right data?, the secure case notes might be distributed across the time range or sort criteria.  If a suitable algorithm can be found - great, but we can work on this on subsequent commits.

Reply:
:+1: